react-app/components/FileManager/utils.js

```diff
@@ -177,3 +178,25 @@ export const getFilesFromATree = (node) => {
 
   return files;
 };
+
+export const getMetadataFiles = (node) => {
+  const assetKeys = [];
+
+  const { name, path, metadata } = node;
+  if (metadata && metadata.asset_key) {
+    assetKeys.push({
+      name: name || '',
+      path: path || '',
+      asset_key: metadata.asset_key || '',
+      url: metadata.external_url || ''
+    });
+  }
+
+  if (node.children && node.children.length) {
+    node.children.forEach((child) => {
+      assetKeys.push(...getMetadataFiles(child));
+    });
+  }
+
+  return assetKeys;
```

Comment:
```suggestion
```suggestion
/**
 * Recursively extracts asset keys from a node object and its children.
 * @param {object} node - The node object containing asset information.
 * @param {string} [node.name] - The name of the node.
 * @param {string} [node.path] - The path of the node.
 * @param {object} [node.metadata] - The metadata object containing asset_key and external_url.
 * @param {string} [node.metadata.asset_key] - The asset key associated with the node.
 * @param {string} [node.metadata.external_url] - The external URL associated with the node.
 * @returns {Array<object>} - An array of objects containing asset keys and related information.
 */
export const getMetadataFromFiles = (node) => {
  const filesMetadata = [];

  const { name, path, metadata } = node;
  if (metadata && metadata.asset_key) {
    filesMetadata.push({
      name: name || '',
      path: path || '',
      asset_key: metadata.asset_key || '',
      url: metadata.external_url || ''
    });
  }

  if (node.children && node.children.length) {
    node.children.forEach((child) => {
      filesMetadata.push(...getMetadataFromFiles(child));
    });
  }
  
  return filesMetadata;
```
```